src/main/java/pt/dinis/common/messages/AuthenticatedMessage.java

```diff
@@ -0,0 +1,44 @@
+package pt.dinis.common.messages;
+
+import java.util.Optional;
+
+/**
+ * Created by tiago on 16-02-2017.
+ */
+public class AuthenticatedMessage implements GenericMessage {
+
+     private GenericMessage message;
+     private String token;
+
+     public AuthenticatedMessage(GenericMessage message, String token) {
+          if (message == null) {
+               throw new IllegalArgumentException();
+          }
+          this.message = message;
+          this.token = token;
+     }
+
+     public GenericMessage getMessage() {
+          return message;
+     }
+
+     public String getToken() {
+          return token;
+     }
+
+     public boolean isAuthenticated() {
+          return (token != null);
+     }
+
+     @Override
+     public Direction getDirection() {
+          return message.getDirection();
+     }
```

Comment:
Não, isto é a sua implementação.
message é um objeto que está dentro deste objeto, este objeto seria chamado por this
se preferires eu posso mudar de message para innerMessage ou notAuthenticatedMessage